src/main/java/org/opensearch/security/configuration/SettingsPermissionValveImpl.java

```diff
@@ -0,0 +1,206 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.configuration;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
+import org.opensearch.action.admin.indices.settings.put.UpdateSettingsRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.security.auditlog.AuditLog;
+import org.opensearch.security.privileges.PrivilegesEvaluationContext;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.securityconf.impl.v7.RoleV7;
+import org.opensearch.security.support.WildcardMatcher;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class SettingsPermissionValveImpl implements SettingsPermissionValve {
+    private static final Logger log = LogManager.getLogger(SettingsPermissionValveImpl.class);
+
+    private final AtomicReference<SecurityDynamicConfiguration<RoleV7>> rolesConfiguration = new AtomicReference<>();
+    private final ClusterService clusterService;
+    private final AdminDNs adminDNs;
+    private final AuditLog auditLog;
+
+    public SettingsPermissionValveImpl(
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        AdminDNs adminDNs,
+        AuditLog auditLog
+    ) {
+        this.clusterService = clusterService;
+        this.adminDNs = adminDNs;
+        this.auditLog = auditLog;
+
+        // Add listener for configuration updates
+        clusterService.addListener(event -> {
+            SecurityDynamicConfiguration<RoleV7> config = rolesConfiguration.get();
+            if (config != null) {
+                // Handle any cluster state related updates if needed
+                // TODO: is this needed here? It's used in DlsFlsValveImpl but seems to be working well without it here on cluster updates
+            }
+        });
+    }
+
+    @Override
+    public boolean invoke(PrivilegesEvaluationContext context, ActionListener<?> listener) {
+        final ActionRequest request = context.getRequest();
+        
+        // Skip validation for admin users
+        if (adminDNs.isAdmin(context.getUser())) {
+            return true;
+        }
+
+        try {
+            if (request instanceof ClusterUpdateSettingsRequest) {
```

Comment:
I would recommend to add an interface to core which marks requests which operate on settings. That could be similar to the `IndicesRequest` interface:

https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/IndicesRequest.java

Otherwise, there is an elevated risk that developers of new actions are not aware of this handling whenever actions are introduced which handle settings.